### PR TITLE
This module exploits a directory traversal vulnerability found in Easy File Sharing FTP Server Version 3.6 and Earlier.

### PR DESCRIPTION
**Pull request by Ahmed-Elhady-Mohamed**
This module exploits a directory traversal vulnerability found in Easy File Sharing FTP Server Version 3.6 and Earlier.

### CODE REVIEW

File: modules/auxiliary/scanner/ftp/easy_file_sharing_ftp.rb [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      },
+      'Platform'       => 'win',
+      'Author'         =>
+        [
+          'Ahmed Elhady Mohamed'   # @kingasmk ahmed.elhady.mohamed[at]gmail.com
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'CVE', '2017-6510']
```
Comment (wchen-r7) [inline modules/auxiliary/scanner/ftp/easy_file_sharing_ftp.rb]: This CVE is unknown to cvedtails.... is there an advisory somewhere?

File: modules/auxiliary/scanner/ftp/easy_file_sharing_ftp.rb [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::Ftp
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::Scanner
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Easy File Sharing FTP Server Version 3.6 and Earlier Directory Traversal Information Disclosure',
```
Comment (wchen-r7) [inline modules/auxiliary/scanner/ftp/easy_file_sharing_ftp.rb]: It seems the website only has version 3.5:
http://www.efssoft.com/ftpserver.htm

Where do you get version 3.6?

File: modules/auxiliary/scanner/ftp/easy_file_sharing_ftp.rb (hunk 1 above)
Comment (Ahmed-Elhady-Mohamed) [inline modules/auxiliary/scanner/ftp/easy_file_sharing_ftp.rb]: I contacted mitre and the assign this cve number for this vulnerability

File: modules/auxiliary/scanner/ftp/easy_file_sharing_ftp.rb (hunk 2 above)
Comment (Ahmed-Elhady-Mohamed) [inline modules/auxiliary/scanner/ftp/easy_file_sharing_ftp.rb]: After installing the 3.5 version. Go to the about box. They mentioned that the installed version is 3.6

Comment (wchen-r7) [inline modules/auxiliary/scanner/ftp/easy_file_sharing_ftp.rb]: Got it. Thanks!

**Comment by wchen-r7** [comment]
Verified, landing soon.

ftp

**Comment by wchen-r7** [comment]
# Release Notes

This adds an auxiliary module that performs a directory traversal attack against a vulnerable Easy File Sharing FTP Server. It allows you to steal files outside the FTP directory.

**Pull request merged**